### DISCUSSION TITLE
Nested accordion animated chevron

### DISCUSSION THREAD

**Discussion by gunnartorfis** [radix-ui/primitives#1070] [Help] [answered]
I'm displaying a nested chevron with, in theory, infinitely many nested headers so it's a recursive rendering. 

Following this example: https://codesandbox.io/s/vkwlb?file=/src/App.js, one can notice that the chevrons aren't displayed inside the nested accordions. When chevrons are added in the nested accordions, they behave incorrectly: https://codesandbox.io/s/radix-nested-accordions-forked-fe11z?file=/src/App.js

It seems like the data-state is correct on the parent divs so I'm not sure how I can animate this correcly. Can someone help me out?

**Comment by benoitgrelard** [comment, marked as answer]
Hi @gunnartorfis,

Yeah this makes sense, it's because this rule is too generic as a `data-state` exists in a few places:
```tsx
"[data-state=open] &": {
  transform: "rotate(180deg)",
  color: theme.colors.violet800.value
}
```

You could solve this by being more specific with what you are targetting:

```diff
-"[data-state=open] &": {
+[`${StyledButton}[data-state=open] &`]: {
  transform: "rotate(180deg)",
  color: theme.colors.violet800.value
}
```

**Reply by gunnartorfis** [reply]
Thank you so much for the help. Works like a charm.
